### PR TITLE
Hot reload proof-of-concept

### PR DESCRIPTION
**Pull request by BeauSLM**
Gets a basic hot reload example going.

- import dependencies and cmake stuff
- parse a toml file
- copy config toml files to build dir
- hot-reload from build dir
- tune in real time with IMGUI
- serialize new config back to vc-controlled file
- instructions for how to try it

### CODE REVIEW

File: code/client/CarPhysics.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#define TOML_EXCEPTIONS 0 // NOTE(beau): I don't want exceptions lmao
+#include "toml++/toml.h"
+
+#define FILEPATH "configs/CarPhysics.toml"
+#define ASSETS_FILEPATH "../../code/assets/configs/CarPhysics.toml"
```
Comment (SamuelGarm) [inline code/client/CarPhysics.h:5]: the serializer doesn't seem to write back to the main version controlled CarPhysics.toml file. I change a value, press t to read it (works fine) then I press s to write it to the main file (the one not in the 'out' folder) and I don't notice any change in the file (even though the console did say it was serializing the right values.
I fixed it by changing the ASSETS_FILEPATH to "../../../../assets/configs/CarPhysics.toml", but I wanted to run it by you before

Comment (BeauSLM) [reply code/client/CarPhysics.h:5]: OH RIGHT aight so your file structure looks different than mine, I believe there's like "out" and "debug" folders in the path to the executable which is why you have to go back two more folders. Here's what my build folder looks like:

image

I'll add some conditional compilation to make it work for the windows filepath - hopefully you and elise have the same build dir structure.

Comment (SamuelGarm) [reply code/client/CarPhysics.h:5]: Sounds good, here is an idea of what my file path looks like to get to the runtime file path
C:\Users\oswei\source\repos\SamuelGarm\CPSC-585\code\out\build\x64-Debug\client\configs
and the regular file path
C:\Users\oswei\source\repos\SamuelGarm\CPSC-585\code\assets\configs

**Comment by BeauSLM** [comment]
NOTE: I've found out that physx 5 provides an example where you deserialize vehicle tunables from a json. I won't work on anymore until I look at that and decide if that approach is worth using.

Additionally, I talked to the TA just now he gave me some insight into his project's tuning setup. They used IMGUI sliders to tune stuff at runtime, and then serialized the values from there. My current setup involves editing a file to tune at runtime, but imo the IMGUI approach is much better at this.

My next iteration of this system will use IMGUI to edit the config in memory, and provide a button to serialize that somehow.

TL;DR we might get this for free with physx and my approach is bad so don't merge just yet.

**Comment by BeauSLM** [comment]
I think this is good enough for now, once we have stuff to debug I'll work on a debug system. Everyone take a look, try it on your machines, and lmk if there's stuff I need to fix.

If there's no issues, I'll merge.

**Comment by BeauSLM** [comment]
works on my machine and Elise's so I merged
